api/src/main/java/com/datastrato/gravitino/Catalog.java

```diff
@@ -7,73 +7,21 @@
 import com.datastrato.gravitino.annotation.Evolving;
 import com.datastrato.gravitino.file.FilesetCatalog;
 import com.datastrato.gravitino.messaging.TopicCatalog;
-import com.datastrato.gravitino.rel.SupportsSchemas;
 import com.datastrato.gravitino.rel.TableCatalog;
-import java.util.Map;
 
 /**
- * The interface of a catalog. The catalog is the second level entity in the gravitino system,
- * containing a set of tables.
+ * The client interface of a catalog. The catalog is the second level entity in the gravitino
+ * system, containing a set of tables. The server side should use the other one with the same name
+ * in the core module.
  */
 @Evolving
-public interface Catalog extends Auditable {
-
-  /** The type of the catalog. */
-  enum Type {
-    /** Catalog Type for Relational Data Structure, like db.table, catalog.db.table. */
-    RELATIONAL,
-
-    /** Catalog Type for Fileset System (including HDFS, S3, etc.), like path/to/file */
-    FILESET,
-
-    /** Catalog Type for Message Queue, like Kafka://topic */
-    MESSAGING,
-
-    /** Catalog Type for test only. */
-    UNSUPPORTED
-  }
-
-  /**
-   * A reserved property to specify the package location of the catalog. The "package" is a string
-   * of path to the folder where all the catalog related dependencies is located. The dependencies
-   * under the "package" will be loaded by Gravitino to create the catalog.
-   *
-   * <p>The property "package" is not needed if the catalog is a built-in one, Gravitino will search
-   * the proper location using "provider" to load the dependencies. Only when the folder is in
-   * different location, the "package" property is needed.
-   */
-  String PROPERTY_PACKAGE = "package";
-
-  /** @return The name of the catalog. */
-  String name();
-
-  /** @return The type of the catalog. */
-  Type type();
-
-  /** @return The provider of the catalog. */
-  String provider();
-
-  /**
-   * The comment of the catalog. Note. this method will return null if the comment is not set for
-   * this catalog.
-   *
-   * @return The comment of the catalog.
-   */
-  String comment();
-
-  /**
-   * The properties of the catalog. Note, this method will return null if the properties are not
-   * set.
-   *
-   * @return The properties of the catalog.
-   */
-  Map<String, String> properties();
+public interface Catalog extends CatalogBasicInfo {
```

Comment:
What is the purpose of having another `CatalogBasicInfo` interface, this seems a bit strange?

Reply:
Because in "Catalog.java", it defines some methods like "asSchema()", "asTableCatalog()" which returns a server-side interface like "SupportsSchema", "TableCatalog"; To separate the "SupportsSchema" interface to two, I also need to separate the Catalog interface into two: one for server side, the other is for client side. 

Today in Catalog.java it also defines some basic attributes like name, type, provider, these information are shared for both server side and client side; So I abstract them into the "CatalogBasicInfo" class. The other catalog interface also extends from it.

Hope this clarifies.

Reply:
I feel like `asXXX` is not used in the server side, so may be we can unify this `Catalog` interface, do not need to separate into two.